internal/terraform/rootmodule/types.go

```diff
@@ -76,7 +76,7 @@ type RootModule interface {
 	MergedSchema() (*schema.BodySchema, error)
 	IsParsed() bool
 	ParseFiles() error
-	ParsedDiagnostics() hcl.Diagnostics
+	ParsedDiagnostics() map[string]hcl.Diagnostics
```

Comment:
@radeksimko I had to track the diagnostics per file, as it played nicer with the LSP.